vehicle_gateway_python/examples/vtol_body_rates.py

```diff
@@ -0,0 +1,218 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import sys
+import time
+
+import vehicle_gateway
+from vehicle_gateway import ControllerType
+
+
+vg = vehicle_gateway.init(args=sys.argv, plugin_type='px4')
+
+TARGET_ALTITUDE = 30  # meters above takeoff point
+
+EARTH_RADIUS = 6378100  # meters
+
+
+def calc_distance_latlon(lat_1, lon_1, lat_2, lon_2):
+    # This uses a planar approximation, not the real trigonometry. This
+    # approximation is just fine for short distances, which is all we're
+    # currently considering.
+    dx = (EARTH_RADIUS * math.pi / 180) * (lon_2 - lon_1) * math.cos(lat_1)
+    dy = (EARTH_RADIUS * math.pi / 180) * (lat_2 - lat_1)
+    return math.sqrt(dx * dx + dy * dy)
+
+
+def clamp(x, max_magnitude):
+    if x > max_magnitude:
+        return max_magnitude
+    elif x < -max_magnitude:
+        return -max_magnitude
+    return x
+
+
+print('Arming...')
+vg.arm_sync()
+time.sleep(2)  # not sure why... perhaps some internal state setting?
+
+home_lat, home_lon, home_alt_amsl = vg.get_latlon()
+
+print('Takeoff!')
+vg.takeoff()
+
+for t in range(0, 10):
+    lat, lon, alt_amsl = vg.get_latlon()
+    dalt = alt_amsl - home_alt_amsl
+    r, p, y = vg.get_euler_rpy()
+    print(f'takeoff delay: {dalt} rpy: {r} {p} {y}')
+    time.sleep(1)
+
+vg.go_to_latlon(home_lat, home_lon, home_alt_amsl + TARGET_ALTITUDE)
+while True:
+    time.sleep(1)
+    lat, lon, alt_amsl = vg.get_latlon()
+    dalt = alt_amsl - home_alt_amsl
+    print(f'mc takeoff climb, current altitude: {dalt}')
+    if dalt > TARGET_ALTITUDE - 5:
+        break  # close enough...
+
+print('Transitioning to fixed-wing...')
+vg.transition_to_fw_sync()
+print(f'VTOL state: {vg.get_vtol_state().name}')
+vg.go_to_latlon(home_lat, home_lon, home_alt_amsl + TARGET_ALTITUDE)
+while True:
+    time.sleep(1)
+    lat, lon, alt_amsl = vg.get_latlon()
+    dalt = alt_amsl - home_alt_amsl
+    r, p, y = vg.get_euler_rpy()
+    print(f'fw transition climbout, alt_amsl: {dalt} rpy: {r} {p} {y}')
+    # wait until we recover to close to our target altitude
+    if dalt > TARGET_ALTITUDE - 2:
+        break  # close enough...
+
+print('begin transitioning to offboard control')
+for t in range(0, 20):
+    time.sleep(0.1)
+    r, p, y = vg.get_euler_rpy()
+    print(f'offboard transition {t} / 200, alt_amsl: {dalt} rpy: {r} {p} {y}')
+    vg.set_body_rates_and_thrust_setpoint(0, 0, 0, 0.7)
+    vg.set_offboard_control_mode(ControllerType.BODY_RATES)
+
+vg.set_offboard_mode()
+
+print('enabled body-rate controller')
+target_north = 300
+target_east = 0
+target_airspeed = 15
+
+lap_count = 0
+while True:
```

Comment:
When I run it, the UAV flies away, which I think is expected, since we're varying thrust. Maybe make this loop happen for 30 seconds and then exit, rather than looking at the distance from a point we're not heading towards, unless I misunderstand something.

Alternatively, maybe we just delete all of the code after this loop, if it's not going to be used.